### PR TITLE
Handle var-args system call arguments for syscall() for RISC-V purecap.

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

File: sys/riscv/riscv/trap.c [hunk 1]
```diff
@@ -123,9 +138,40 @@ cpu_fetch_syscall_args(struct thread *td)
 		sa->callp = &p->p_sysent->sv_table[sa->code];
 
 	sa->narg = sa->callp->sy_narg;
-	memcpy(sa->args, ap, nap * sizeof(syscallarg_t));
-	if (sa->narg > nap)
-		panic("TODO: Could we have more then %d args?", NARGREG);
+#if __has_feature(capabilities)
+	if (stack_args != NULL) {
+		uint64_t intval;
```
Comment (jrtc27) [inline sys/riscv/riscv/trap.c:143]: Strictly speaking we should probably use `register_t`/`fueword` rather than hard-code that it's 64-bit.

Comment (bsdjhb) [reply sys/riscv/riscv/trap.c]: Done.

**Comment by jrtc27** [comment]
I think we want something like that; I had to go look up whether `register_t` was still an integer on RISC-V, because it would be completely sensible to make that be a capability when you have a merged register file.

**Comment by bsdjhb** [comment]
Yes, I'm a bit torn on register_t.  I do kind of think it makes sense for register_t to be a uintcap_t, but that probably breaks several things.  It would permit using register_t instead of syscallarg_t again though which would reduce some diffs to upstream.

**Pull request merged**